File: 2204-find-subsequence-of-length-k-with-the-largest-sum/2204-find-subsequence-of-length-k-with-the-largest-sum.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[]}
def max_subsequence(nums, k)
    arr = []
    nums.each_with_index do |x, i|
        arr << [x, i]
    end
    arr.sort_by! {|x| -x[0]}
    arr = arr.first(k)
    arr.sort_by! {|x| x[1]}
    ans = arr.map {|x| x[0]}
    return ans
end

File: 2204-find-subsequence-of-length-k-with-the-largest-sum/2204-find-subsequence-of-length-k-with-the-largest-sum.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        arr = []
        for i, num in enumerate(nums):
            arr.append([num, i])
        arr.sort(reverse = True)
        arr = arr[:k]
        arr.sort(key=lambda x: x[1])
        ans = [val[0] for val in arr]
        return ans
        